flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/KinesisConfigUtil.java

```diff
@@ -222,19 +224,28 @@ private static void validateOptionalPositiveDoubleProperty(Properties config, St
 		}
 	}
 
-	private static void validateOptionalDateProperty(Properties config, String key, String message) {
-		if (config.containsKey(key)) {
-			try {
-				initTimestampDateFormat.parse(config.getProperty(key));
-			} catch (ParseException parseException) {
+	private static void validateOptionalDateProperty(Properties config, String timestampKey, String formatKey, String message) {
+		if (config.containsKey(timestampKey)) {
+			if (config.containsKey(formatKey)) {
```

Comment:
This here too is a bit too complicated.
We can just determine what format will be used by `config.get(formatKey, DEFAULT)`, and focus on validating the date. I don't think we need to make things complicated by checking if the format key exists.